streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -244,18 +274,117 @@ public static Map<ProcessId, KafkaStreamsAssignment> optimizeRackAwareStandbyTas
         );
         LOG.info("Assignment before standby task optimization has cost {}", initialCost);
 
-        throw new UnsupportedOperationException("Not yet Implemented.");
+        final MoveStandbyTaskPredicate moveablePredicate = getStandbyTaskMovePredicate(applicationState);
+        final BiFunction<KafkaStreamsAssignment, KafkaStreamsAssignment, List<TaskId>> getMovableTasks = (source, destination) -> {
+            return source.tasks().values().stream().filter(task -> task.type() == AssignedTask.Type.STANDBY)
+                .filter(task -> {
+                    return !destination.tasks().containsKey(task.id());
+                })
```

Comment:
```suggestion
                .filter(task -> !destination.tasks().containsKey(task.id()))
```
Just makes it a bit easier to read when we keep lambdas on one line by getting rid of the brackets, especially when we have a complex structure containing multiple steps with lambdas, nested lambda, etc